packages/office-ui-fabric-react/src/components/Coachmark/Coachmark.tsx

```diff
@@ -190,41 +250,53 @@ export class Coachmark extends BaseComponent<ICoachmarkProps, ICoachmarkState> {
      * The target element the mouse would be in
      * proximity to
      */
-    let targetElementRect: ClientRect = targetElement.getBoundingClientRect();
-
-    // When the window resizes we want to async
-    // get the bounding client rectangle.
-    // Every time the event is triggered we wan't to
-    // setTimeout and then clear any previous instances
-    // of setTimeout.
-    this._events.on(window, 'resize', (): void => {
-      timeoutIds.forEach((value: number): void => {
-        clearInterval(value);
-      });
+    let targetElementRect: ClientRect;
 
-      timeoutIds.push(this._async.setTimeout((): void => {
-        targetElementRect = targetElement.getBoundingClientRect();
-      }, 100));
-    });
+    // Take the initial measure out of the initial render to prevent
+    // an unnessecary render.
+    this._async.setTimeout(() => {
+      targetElementRect = this._entityInnerHostElement.getBoundingClientRect();
 
-    // Everytime the document's mouse move is triggered
+      // When the window resizes we want to async
+      // get the bounding client rectangle.
+      // Every time the event is triggered we want to
+      // setTimeout and then clear any previous instances
+      // of setTimeout.
+      this._events.on(window, 'resize', (): void => {
+        timeoutIds.forEach((value: number): void => {
+          clearInterval(value);
+        });
+
+        timeoutIds.push(this._async.setTimeout((): void => {
+          targetElementRect = this._entityInnerHostElement.getBoundingClientRect();
+        }, 100));
+      });
+    }, 10);
+
+    // Every time the document's mouse move is triggered
     // we want to check if inside of an element and
     // set the state with the result.
     this._events.on(document, 'mousemove', (e: MouseEvent) => {
       let mouseY = e.pageY;
       let mouseX = e.pageX;
-      let isMouseInProximity = this._isInsideElement(targetElementRect, mouseX, mouseY);
+      let isMouseInProximity = this._isInsideElement(mouseX, mouseY, targetElementRect, mouseProximityOffset);
 
       if (isMouseInProximity !== this.state.isMouseInProximity) {
         this.setState({
-          isMouseInProximity: isMouseInProximity
```

Comment:
Don't you still need to update isMouseInProximity?

Reply:
No, i'm actually not using it anywhere else.

Reply:
Yes you do, right above, you do a check. `if (isMouseInProximity !== this.state.isMouseInProximity)`

Reply:
Ah, lol not sure why my brain missed that entirely. Adding it back.

Reply:
Wait, the reason i removed this is because we only want to ever open the coachmark and never collapse it once it's been opened. So this was actually correct.

Reply:
Ooohh, can you add a comment then?